packages/aws-cdk/lib/notices.ts

```diff
@@ -0,0 +1,264 @@
+import * as https from 'https';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import * as semver from 'semver';
+import { debug, print } from './logging';
+import { cdkCacheDir } from './util/directories';
+import { versionNumber } from './version';
+
+const CACHE_FILE_PATH = path.join(cdkCacheDir(), 'notices.json');
+
+export interface DisplayNoticesProps {
+  /**
+   * The cloud assembly directory. Usually 'cdk.out'.
+   */
+  readonly outdir: string;
+
+  /**
+   * Issue numbers of notices that have been acknowledged by a user
+   * of the current CDK repository. These notices will be skipped.
+   */
+  readonly acknowledgedIssueNumbers: number[];
+
+  /**
+   * Whether cached notices should be ignored. Setting this property
+   * to true will force the CLI to download fresh data
+   *
+   * @default false
+   */
+  readonly ignoreCache?: boolean;
+}
+
+export async function refreshNotices() {
+  const dataSource = dataSourceReference(true);
```

Comment:
I don't think this necessarily MUST ignore the cache... just that IF we're going to refresh, we'd better do it early so that we don't add latency later. It's perfectly fine not to refresh here if the data is not stale yet.